src/lib/comms/sol-http-common.c

```diff
@@ -843,15 +844,27 @@ sol_http_split_query(const char *query, struct sol_http_params *params)
         }
 
         if (!sol_http_param_add_copy(params,
-            CREATE_QUERY_PARAM(key, value))) {
+            CREATE_PARAM(key, value))) {
             SOL_ERR("Could not add the HTTP param %.*s:%.*s",
                 SOL_STR_SLICE_PRINT(key), SOL_STR_SLICE_PRINT(value));
             goto exit;
         }
     }
-#undef CREATE_QUERY_PARAM
+#undef CREATE_PARAM
 
 exit:
     sol_vector_clear(&tokens);
     return 0;
 }
+
+SOL_API int
+sol_http_split_query(const char *query, struct sol_http_params *params)
+{
+    return sol_http_split_str_key_value(query, SOL_HTTP_PARAM_QUERY_PARAM, params);
+}
```

Comment:
do you think is better make `sol_http_split_str_key_value()` public ?

I did not do it to avoid mistakes, this function has hardcoded the delimiters `&` and `=` and it will not work for all key/value types, for instance, headers use `:`.

I think to make it public it will be necessary more flexible, to my mind is better gives those specific functions like `sol_http_split_query()`